dumux-precice/couplingadapter.cc

```diff
@@ -162,47 +131,16 @@ double CouplingAdapter::getScalarQuantityOnFace(const size_t dataID,
             "created!");
     }
     const auto idx = indexMapper_.getPreciceId(faceID);
-    assert(dataID < dataVectors_.size());
-    const std::vector<double> &quantityVector = dataVectors_[dataID];
-    assert(idx < quantityVector.size());
-    return quantityVector[idx];
+    std::vector<double> &dataVector = getQuantityVector(meshName, dataName);
```

Comment:
Why do we call `getQuantityVector` inside the function `getScalarQuantityOnFace`? Is it because we consider multiple faces?

Reply:
This is in the same logic with Alexander's design: get the vector for this data first, then find the value for a specific face.